README.md

```diff
@@ -0,0 +1,224 @@
+# xtm - a library for cross-thread messaging with event loop integration
+
+The library provides the following facilities:
+
+# xtm_queue
+
+Opaque struct, that represent unidirectional, single-writer-single-reader queue
+implementation with event loop integration ability.
+
+## xtm_queue_new
+
+Allocation function, used to create instance of `struct xtm_queue` and return
+its pointer or `NULL` in case of error. Accepts the queue size as input, which
+must be a power of two, and flags, defining library behavior.
+
+## xtm_queue_delete
+
+Deallocation function, used to free queue and close its internal fds, in case
+when appropriate flag (`XTM_QUEUE_MUST_CLOSE_READFD`) is set, when queue was
+allocated.
+Returns 0 on success. Otherwise -1 with errno set appropriately (as in
+`close(2)`, since it implies close of fds)
+
+## xtm_queue_notify_consumer
+
+Function for queue consumer notification.
+Returns 0 on success. Otherwise -1 with errno set appropriately (as in
+`write(2)`, since it implies write to internal fd)
+
+## xtm_queue_notify_producer
+
+Function for queue producer notification.
+Returns 0 on success. Otherwise -1 with errno set appropriately (as in
+`write(2)`, since it implies write to internal fd)
+
+## xtm_queue_probe
+
+Returns 0 if queue has space. Otherwise -1 with errno set to `ENOBUFS`.
+
+## xtm_queue_count
+
+Function returns current message count in queue.
+
+## xtm_queue_push_fun
+
+Function push function and its argument to the queue. This function does not
+notify the consumer thread, but only push to the queue. To notify the consumer
+thread you must call `xtm_queue_notify_consumer`. The less often you notify,
+the greater the performance, but the greater the latency.
+Function accepts flags, which defined it's behaviour:
+XTM_QUEUE_PRODUCER_NEEDS_NOTIFICATIONS - if this flag is set, producer sets,
+special flag in queue internals, if push fails. Consumer can check this flag,
+using `xtm_queue_get_reset_was_full` function and notifies producer.
+Returns 0 if queue has space. Otherwise -1 with errno set to `ENOBUFS`.
+
+## xtm_queue_consumer_fd
+
+Returns file descriptor, that should be watched by consumer thread to
+become readable. When it became readable, consumer should call one of
+the consumer functions: `xtm_queue_pop_ptrs` or `xtm_queue_invoke_funs_all`.
+
+## xtm_queue_producer_fd
+
+Return file descriptor, that should be watched by producer thread to become
+readable in case when `xtm_queue_push_fun` or `xtm_queue_push_ptr` fails. When
+it became readable, producer may try push to the queue again. Very rarely,
+there may be a situation (because of race between producer and consumer
+threads), when the descriptor became readable, but queue is still full. In this
+case producer thread, need to poll this descriptor again.
```

Comment:
This comma is clearly excess. Also need -> needs.

Reply:
Fixed